compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -1798,21 +1796,28 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
 
         let type_const_args = self.collect_type_and_const_args(&ast_generics.params);
 
-        let universal_args: Vec<_> = universal_paths
-            .into_iter()
-            .map(|ty| match ty {
-                hir::TyKind::Path(hir::QPath::Resolved(ty, path @ hir::Path { span, .. })) => {
-                    hir::GenericArg::Type(hir::Ty {
-                        hir_id: self.next_id(),
-                        span: *span,
-                        kind: hir::TyKind::Path(hir::QPath::Resolved(ty, path)),
-                    })
-                }
-                _ => {
-                    unreachable!("{:?} should be of type TyKind::Path(..)", ty);
-                }
-            })
-            .collect();
+        let universal_args: Vec<_> = impl_trait_inputs.iter().map(|ty| match ty.kind {
+            TyKind::ImplTrait(node_id, _) => {
+                let def_id = self.resolver.local_def_id(node_id);
```

Comment:
notably: this points to the _original_ def id, not the new def id. that's the opaque type that actually shows up in the body of the function, so if we pass in the new def id, then we'll be defining `APIT: Sized` on the _wrong_ APIT (the RPIT's copy of the APIT).